scheduler/scheduler.go

```diff
@@ -147,11 +152,22 @@ func (scheduler *Scheduler) SystemStart() {
 }
 
 func (scheduler *Scheduler) Schedule(schedTime uint) {
+	var tick <-chan time.Time = time.Tick(time.Duration(scheduler.checkTime) * time.Millisecond)
 	for {
 		time.Sleep(time.Millisecond * time.Duration(schedTime))
 		common.LogDebug(common.Debug, "System is using", scheduler.usedCores, "cores now.", uint8(len(scheduler.freeCores))-scheduler.usedCores, "cores are left available.")
 		for i := range scheduler.Clonable {
 			ff := scheduler.Clonable[i]
+			select {
+			// After checkTime milliseconds we suppose that flow function behaviour changed and want
+			// to try to add some clones (if can). If adding clones decrease performance we will
+			// stop them and wait for another checkTime period. If we know speed with more clones
+			// we forget it.
```

Comment:
Regular flow function speed check after each checkTime milliseconds, it is done by erasing previously measured speed data. If it is possible, scheduler adds some new clones. If performance decreases with more added clones, they are stopped until another checkTime milliseconds pass.